### PR TITLE
benchmark: allow setting seed on the CLI

### PR DESCRIPTION
**Pull request by cb22**
For now, it's still fixed. We might want to randomize this though with some future thought (or perhaps only combined with --verify in future).

Additionally, sets the account user_data_* fields to random values too.

### CODE REVIEW

**Comment by matklad** [comment]
Hm, I'd say for benchmarks we do want the seed to be the same by default? It's easier to compare benchmarks if you know that the code is exactly the same. Opt-in randomization might be a useful thing though

**Comment by cb22** [comment]
> Hm, I'd say for benchmarks we do want the seed to be the same by default? It's easier to compare benchmarks if you know that the code is exactly the same. Opt-in randomization might be a useful thing though

Yeah it's a valid point, I'm kinda blending this as part of making the benchmark do end-to-end verification, and there it's useful to have unique values each time - I might just swap the default here.

I've got another follow-up PR that actually allows you to disable the attempt at targeting a specific rate, to get even more deterministic batches. Currently, it's actually pretty random what happens in the warm-up phase of the benchmark.

As a counterpoint to the first point though, I would - at least glancing at the code - not expect the seed to affect the results that much (barring pending transfers - but that's disabled by default) since the shape of the data is still the same, ~but that's _not_ the case - I've got 2 seeds that have a ~50k TPS delta between them which is far more than expected.~ (maybe not...)

File: src/tigerbeetle/benchmark_load.zig [hunk 1]
```diff
@@ -223,9 +233,9 @@ const Benchmark = struct {
         {
             b.batch_accounts.appendAssumeCapacity(.{
                 .id = b.account_id_permutation.encode(b.account_index + 1),
-                .user_data_128 = 0,
-                .user_data_64 = 0,
-                .user_data_32 = 0,
+                .user_data_128 = random.int(u128),
+                .user_data_64 = random.int(u64),
+                .user_data_32 = random.int(u32),
```
Comment (batiati) [inline src/tigerbeetle/benchmark_load.zig:234]: ❤️

File: src/tigerbeetle/benchmark_load.zig [hunk 2]
```diff
@@ -129,7 +129,15 @@ pub fn main(
         );
     }
 
-    var rng = std.rand.DefaultPrng.init(42);
+    const seed: usize = cli_args.seed orelse seed: {
+        // If no seed was given, use a random seed instead.
+        var seed_random: u64 = undefined;
+        try std.os.getrandom(std.mem.asBytes(&seed_random));
+        break :seed seed_random;
+    };
```
Comment (batiati) [inline src/tigerbeetle/benchmark_load.zig]: Do you want to do it here or in another PR?

```suggestion
    // If no seed was given, use a default seed for reproducibility.
    const seed: usize = cli_args.seed orelse 42;
```

Comment (cb22) [reply src/tigerbeetle/benchmark_load.zig]: I'll do it here - I'll commit and rebase this once I'm finished digging into my strange results!

Comment (cb22) [reply src/tigerbeetle/benchmark_load.zig]: @batiati done!

**Review by batiati — APPROVED** [review]
